src/Entity/Document.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Pucene\Bundle\DoctrineBundle\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+
+/**
+ * Document entity.
+ */
+class Document
```

Comment:
@wachterjohannes for me definitly not as this class is not needed to be in the core. 

For me the architecture should be the following:

There is a the pucene bundle which implements query builder, read type configuration from xml, yml or annotation, mapping and analyze. There is a simple config parameter `storage` which you can use the `raw`, `elasticsearch`, `mysql`. 

The storage bundle itself need to implement the StorageInterface.

```
public function request($json, $method)
```

The core or elasticsearch storage dont need any Index, Document, Mapping, Field or Token entities with relation to each others they are just needed for the doctrine implemention to make queries.